pkg/cmd/server/origin/handlers.go

```diff
@@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
 	})
 }
 
+// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
+// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
+func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
+	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
+	if skewedClientPolicy == configapi.AllowAll {
+		return handler
+	}
+	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
```

Comment:
It's fine to have logic in both spots for now, we just need to have an
upstream issue to avoid having to parse.  No one ever spent enough time on
this, really.

On Wed, Mar 2, 2016 at 5:12 PM, David Eads <notifications@github.com> wrote:

> In pkg/cmd/server/origin/handlers.go
> <https://github.com/openshift/origin/pull/7741#discussion_r54809491>:
>
> > @@ -168,6 +176,98 @@ func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMa
> >  	})
> >  }
> >
> > +// versionSkewFilter adds a filter that may deny PUT an POST requests since they stomp entire resource objects from skewed
> > +// oc clients, since we know that those clients will strip unknown fields which can lead to unexpected outcomes
> > +func (c *MasterConfig) versionSkewFilter(binaryInfo version.Info, handler http.Handler) http.Handler {
> > +	skewedClientPolicy := c.Options.PolicyConfig.SkewedClientPolicy
> > +	if skewedClientPolicy == configapi.AllowAll {
> > +		return handler
> > +	}
> > +	seg := strings.SplitN(binaryInfo.GitVersion, "-", 2)
>
> This is done upstream to build the user agent string. Change it there too?
> It would be a lot easier to do if the type wasn't unversioned.
>
> So, same logic in both spots, new field duplicating information, or
> replumb the logic through both sides discrete fields?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7741/files#r54809491>.
>